onnx/defs/math/defs.cc

```diff
@@ -505,24 +505,42 @@ ONNX_OPERATOR_SET_SCHEMA(
             "Constrain input and output types to float tensors.")
         .TypeAndShapeInferenceFunction(propagateShapeAndTypeFromFirstInput));
 
-static const char* Pow_ver7_doc = R"DOC(
+static const char* Pow_ver12_doc = R"DOC(
 Pow takes input data (Tensor<T>) and exponent Tensor, and
 produces one output data (Tensor<T>) where the function `f(x) = x^exponent`,
 is applied to the data tensor elementwise.
 )DOC";
 
 ONNX_OPERATOR_SET_SCHEMA(
     Pow,
-    7,
+    12,
     OpSchema()
-        .SetDoc(std::string(Pow_ver7_doc) + GenerateBroadcastingDocMul())
+        .SetDoc(std::string(Pow_ver12_doc) + GenerateBroadcastingDocMul())
         .Input(0, "X", "First operand, base of the exponent.", "T")
-        .Input(1, "Y", "Second operand, power of the exponent.", "T")
+        .Input(1, "Y", "Second operand, power of the exponent.", "T1")
         .Output(0, "Z", "Output tensor (same size as X)", "T")
         .TypeConstraint(
             "T",
-            {"tensor(float16)", "tensor(float)", "tensor(double)"},
-            "Constrain input and output types to float tensors.")
+            {"tensor(int32)",
```

Comment:
The set of allowed types T is more limited than T1 for exponent. Is there a strong reason not to support unsigned integers?

Reply:
I restricted the types of T following this comment https://github.com/onnx/onnx/pull/2666#discussion_r394071964 to avoid overflow.
Another option would be to accept any type for X as well. 
But then the output type would be impossible to infer..